linalg/hypre.cpp

```diff
@@ -1131,7 +1132,7 @@ HypreParMatrix::HypreParMatrix(
 HypreParMatrix::HypreParMatrix(MPI_Comm comm,
                                HYPRE_BigInt *row_starts,
                                HYPRE_BigInt *col_starts,
-                               SparseMatrix *sm_a)
+                               const SparseMatrix *sm_a)
```

Comment:
The `const` here is correct, however, in order to call `CopyCSR`, we'll need to `const_cast` away the `const`-ness which is fine since we know that we do not actually modify any of the arrays in `sm_a` later using `mem_a`.